code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,598 @@
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	if(active_shield)
+		if(plasma_cost > xeno_owner.plasma_stored)
+			owner.balloon_alert(owner, "[plasma_cost - xeno_owner.plasma_stored] more plasma!")
+			return FALSE
+		if(can_use_action(FALSE, XACT_USE_BUSY))
+			shield_blast()
+			cancel_shield()
+		return
+
+	if(A)
+		owner.dir = get_cardinal_dir(owner, A) //if activated by mouse click, we face the atom clicked
+
+	var/turf/target_turf = get_step(owner, owner.dir)
+	if(target_turf.density)
+		owner.balloon_alert(owner, "Obstructed by [target_turf]")
+		return
+	for(var/atom/movable/affected AS in target_turf)
+		if(affected.density)
+			owner.balloon_alert(owner, "Obstructed by [affected]")
+			return
+
+	succeed_activate()
+	playsound(owner,'sound/effects/magic.ogg', 75, 1)
+
+	action_icon_state = "psy_shield_reflect"
+	xeno_owner.update_glow(3, 3, "#5999b3")
+
+	GLOB.round_statistics.psy_shields++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "psy_shields")
+
+	active_shield = new(target_turf, owner)
+	if(!do_after(owner, 6 SECONDS, TRUE, owner, BUSY_ICON_DANGER, extra_checks = CALLBACK(src, .proc/can_use_action, FALSE, XACT_USE_BUSY)))
+		cancel_shield()
+		return
+	cancel_shield()
+
+///Removes the shield and resets the ability
+/datum/action/xeno_action/activable/psychic_shield/proc/cancel_shield()
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	action_icon_state = "psy_shield"
+	xeno_owner.update_glow()
+	add_cooldown()
+	if(active_shield)
+		active_shield.release_projectiles()
+		QDEL_NULL(active_shield)
+
+///AOE knockback triggerable by ending the shield early
+/datum/action/xeno_action/activable/psychic_shield/proc/shield_blast()
+	succeed_activate()
+
+	active_shield.reflect_projectiles()
+
+	owner.visible_message(span_xenowarning("[owner] sends out a huge blast of psychic energy!"), span_xenowarning("We send out a huge blast of psychic energy!"))
+
+	var/turf/lower_left
+	var/turf/upper_right
+	switch(active_shield.dir)
+		if(NORTH)
+			lower_left = locate(owner.x - 1, owner.y + 1, owner.z)
+			upper_right = locate(owner.x + 1, owner.y + 2, owner.z)
+		if(SOUTH)
+			lower_left = locate(owner.x - 1, owner.y - 2, owner.z)
+			upper_right = locate(owner.x + 1, owner.y - 1, owner.z)
+		if(WEST)
+			lower_left = locate(owner.x - 2, owner.y - 1, owner.z)
+			upper_right = locate(owner.x - 1, owner.y + 1, owner.z)
+		if(EAST)
+			lower_left = locate(owner.x + 1, owner.y - 1, owner.z)
+			upper_right = locate(owner.x + 2, owner.y + 1, owner.z)
+
+	for(var/turf/affected_tile AS in block(lower_left, upper_right)) //everything in the 2x3 block is found.
+		affected_tile.Shake(4, 4, 2 SECONDS)
+		for(var/atom/movable/affected in affected_tile)
+			if(!ishuman(affected) && !istype(affected, /obj/item) && !isdroid(affected))
+				affected.Shake(4, 4, 20)
+				continue
+			if(ishuman(affected))
+				var/mob/living/carbon/human/H = affected
+				if(H.stat == DEAD)
```

Comment:
Probably shouldn't do anything if you are unconcious as well

Reply:
Why not? Other xeno CC works on crit ungas.